crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -271,35 +273,37 @@ where
             flox_nix_content_with_new_packages(&current_manifest_contents, packages)?;
         match new_manifest_contents {
             ManifestContent::Unchanged => return Ok(false),
-            ManifestContent::Changed(new_manifest_contents) => {
-                let mut temp_env = self.make_temporary()?;
-                temp_env.update_manifest(&new_manifest_contents)?;
-                temp_env.build(nix, system).await?;
-                self.replace_with(temp_env)?;
+            ManifestContent::Changed(contents) => {
+                self.transact_with_manifest_contents(contents, nix, system)
+                    .await?;
                 Ok(true)
```

Comment:
since the error type is the same you could do:
```suggestion
            ManifestContent::Changed(contents) => {
                self.transact_with_manifest_contents(contents, nix, system)
                    .await
                    .map(|_| true)
```

I think the way you have it is fine too, at least for me easier to follow.

Reply:
I don't personally like the `|_|` and `.and(Ok(true))` doesn't seem that nice either. But I'll change it since you find it easier to follow 🤷

Reply:
no, i liked yours better :D

Reply:
Oh lol okay I'll leave it